cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/SourceControlMetadataStore.java

```diff
@@ -0,0 +1,228 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.store;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import io.cdap.cdap.common.ApplicationNotFoundException;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.sourcecontrol.SourceControlMeta;
+import io.cdap.cdap.spi.data.StructuredRow;
+import io.cdap.cdap.spi.data.StructuredTable;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.TableNotFoundException;
+import io.cdap.cdap.spi.data.table.field.Field;
+import io.cdap.cdap.spi.data.table.field.Fields;
+import io.cdap.cdap.spi.data.table.field.Range;
+import io.cdap.cdap.store.StoreDefinition;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * Store for namespace and repository source control metadata.
+ */
+public class SourceControlMetadataStore {
+
+  private StructuredTable namespaceSourceControlMetadataTable;
+  private final StructuredTableContext context;
+
+  public static SourceControlMetadataStore create(StructuredTableContext context) {
+    return new SourceControlMetadataStore(context);
+  }
+
+  private SourceControlMetadataStore(StructuredTableContext context) {
+    this.context = context;
+  }
+
+  private StructuredTable getNamespaceSourceControlMetadataTable() {
+    try {
+      if (namespaceSourceControlMetadataTable == null) {
+        namespaceSourceControlMetadataTable = context.getTable(
+            StoreDefinition.NamespaceSourceControlMetadataStore.NAMESPACE_SOURCE_CONTROL_METADATA);
+      }
+    } catch (TableNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    return namespaceSourceControlMetadataTable;
+  }
+
+  /**
+   * Retrieves the source control metadata for the specified application ID in the namespace from
+   * {@code NamespaceSourceControlMetadata} table.
+   *
+   * @param appId {@link ApplicationId} for which the source control metadata is being retrieved.
+   * @return The {@link SourceControlMeta} associated with the application ID, or {@code null} if no
+   *         metadata is found.
+   * @throws IOException If it fails to read the metadata.
+   */
+  @Nullable
+  public SourceControlMeta getNamespaceSourceControlMeta(ApplicationId appId) throws IOException {
```

Comment:
don't need Namespace or SourceControlMeta in the method name, this can just be `get(ApplicationId appId)`. No namespace because this is actually getting info for an application, and no need for SourceControlMeta because the class is already named SourceControlMetadataStore so it would be redundant.

Similar comment for all the other methods.

Reply:
Noted. Done